tests/validation/domain_test.go

```diff
@@ -0,0 +1,41 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/sclevine/spec"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/rancher/rio/tests/testutil"
+)
+
+func domainTests(t *testing.T, when spec.G, it spec.S) {
+
+	var service testutil.TestService
+	var domain testutil.TestDomain
+
+	it.After(func() {
+		service.Remove()
+	})
+
+	when("get a cluster domain", func() {
+		it.Before(func() {
+			service.Create(t, "strongmonkey1992/autoscale:testing")
+		})
+
+		it.After(func() {
+			domain.UnRegister()
+		})
+
+		it("should create a new DNS CNAME for the cluster domain and retrieve the service content using it", func() {
+			clusterDomain := service.GetKubeFirstClusterDomain()
```

Comment:
So to summarize this test, we are grabbing the ClusterDomain and then sending that to route53 which points a CNAME specified in an env var to it, and then lower we are querying that cname ?

Reply:
Exactly.